apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx

```diff
@@ -98,163 +129,213 @@ const PostgrestConfig = () => {
   const isGraphqlExtensionEnabled =
     (extensions ?? []).find((ext) => ext.name === 'pg_graphql')?.installed_version !== null
 
-  return (
-    <Form id={formId} initialValues={initialValues} validate={() => {}} onSubmit={updateConfig}>
-      {({ handleReset, resetForm, values, initialValues }: any) => {
-        const hasChanges = JSON.stringify(values) !== JSON.stringify(initialValues)
-        // [Alaister] although this "technically" is breaking the rules of React hooks
-        // it won't error because the hooks are always rendered in the same order
-        // eslint-disable-next-line react-hooks/rules-of-hooks
-        useEffect(() => {
-          if (config) {
-            const values = {
-              db_schema: config.db_schema,
-              max_rows: config.max_rows,
-              db_extra_search_path: config.db_extra_search_path ?? '',
-              db_pool: config.db_pool || null,
-            }
+  function resetForm() {
+    if (config) {
+      form.reset({
+        dbSchema: config.db_schema ? config.db_schema.replace(/ /g, '').split(',') : [],
+        maxRows: config.max_rows,
+        dbExtraSearchPath: config.db_extra_search_path,
+        dbPool: config.db_pool,
+      })
+    }
+  }
 
-            resetForm({ values, initialValues: values })
+  useEffect(() => {
```

Comment:
can you use the isSuccess callback here like Ivan suggested for that other useEffect issue?

Reply:
Can we change this in a follow up PR. Didn't want to get involved in changing how the form resets in this one.